tensorflow/tools/snapcraft/snap/snapcraft.yaml

```diff
@@ -0,0 +1,125 @@
+ 
+grade: stable
+confinement: classic
+base: core18
+
+architectures:  
+  - build-on: amd64
+
+parts:
+  tensorflow:
+    source: https://github.com/tensorflow/tensorflow.git
+    source-tag: v1.12.0
+    plugin: python
+    python-version: python3
+    build-packages:
+      - build-essential
+      - software-properties-common
+      - zlib1g-dev
+      - libzmq3-dev
+      - libhdf5-serial-dev
+      - libfreetype6-dev 
+      - libcurl3-dev
+      - python3-pip
+      - python3-numpy
+ 
+ tensorflow-gpu:
+    plugin: cmake
+    source:
+      # Tensorflow GPU is only available for amd64 (x86_64),
+      # so fail the build on other architectures.
+      - on amd64: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.0.130-1_amd64.deb
+      
+    source-type: deb
+    override-build: |
+      ARCHITECTURE=$(dpkg --print-architecture)
+      if [ "${ARCHITECTURE}" = "amd64" ]; then
+        DEB_API="https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-repo-ubuntu1804_10.0.130-1_amd64.deb"
+      else
+        echo "ERROR! Tensorflow GPU only produces debs for amd64. Failing the build here."
+        exit 1
+      fi
+      DEB_URL=$(curl -w "%{url_effective}\n" -I -L -s -S "${DEB_API}" -o /dev/null)
+      VERSION=$(echo "${DEB_URL}" | cut -d'_' -f2)
+      echo $VERSION > $SNAPCRAFT_STAGE/version
+      
+      snapcraftctl build
+      
+    build-packages:
+      - curl
+      - dpkg
+      - sed
+      - wget
+      
+    stage-packages:
+      - libnvidia-cfg1-390
+      - libnvidia-cfg1-418
+      - libnvidia-common-390
+      - libnvidia-common-418
+      - libnvidia-compute-390
+      - libnvidia-compute-418
+      - libnvidia-decode-390
+      - libnvidia-decode-418
+      - libnvidia-encode-390
+      - libnvidia-encode-418
+      - libnvidia-fbc1-390
+      - libnvidia-fbc1-418
+      - libnvidia-gl-390
+      - libnvidia-gl-418
+      - libnvidia-ifr1-390
+      - libnvidia-ifr1-418
+      - libvdpau1
+      - libxnvctrl0
+      - mesa-vdpau-drivers
+      - nvidia-compute-utils-390
+      - nvidia-compute-utils-418
+      - nvidia-dkms-390
+      - nvidia-dkms-418
+      - nvidia-driver-390
+      - nvidia-driver-410
+      - nvidia-driver-418
+      - nvidia-kernel-common-390
+      - nvidia-kernel-common-418
+      - nvidia-kernel-source-390
+      - nvidia-kernel-source-418
+      - nvidia-prime
+      - nvidia-settings
+      - nvidia-utils-390
+      - nvidia-utils-418
+      - vdpau-driver-all
+      - xserver-xorg-video-nvidia-390
+      - xserver-xorg-video-nvidia-418
+
+    prime:
+      -usr/share/doc
+      -usr/share/fonts
+      -usr/share/icons
+      -usr/share/lintian
+      -usr/share/man
+
+  wrapper:
+    after:
+      - tensorflow
+    plugin: cmake
+    source: files/
+
+apps:
+ tensorflow:
+    command: bin/wrapper $SNAP/usr/share/tensorflow/bin/tensorflow
```

Comment:
What is this? there is no "tensorflow" binary, its a library. also what does the .desktop file run?

and the url-handler below doesnt make sense either? theres no graphical program to run

Reply:
Hi @perfinion, I forgot to remove those components, but I have removed them now